lib/api/model/model.dart

```diff
@@ -31,6 +31,82 @@ class CustomProfileField {
   Map<String, dynamic> toJson() => _$CustomProfileFieldToJson(this);
 }
 
+@JsonSerializable(fieldRename: FieldRename.snake)
+class ProfileFieldUserData {
+  final String value;
+  final String? renderedValue;
+
+  ProfileFieldUserData({required this.value, this.renderedValue});
+
+  factory ProfileFieldUserData.fromJson(Map<String, dynamic> json) =>
+    _$ProfileFieldUserDataFromJson(json);
+
+  Map<String, dynamic> toJson() => _$ProfileFieldUserDataToJson(this);
+}
+
+/// As in [InitialSnapshot.realmUsers], [InitialSnapshot.realmNonActiveUsers], and [InitialSnapshot.crossRealmBots].
+///
+/// In the Zulip API, the items in realm_users, realm_non_active_users, and
+/// cross_realm_bots are all extremely similar. They differ only in that
+/// cross_realm_bots has is_system_bot.
+///
+/// https://zulip.com/api/register-queue#response
+@JsonSerializable(fieldRename: FieldRename.snake)
+class User {
+  final int userId;
+  @JsonKey(name: 'delivery_email')
+  String? deliveryEmailStaleDoNotUse; // TODO see [RealmUserUpdateEvent.deliveryEmail]
+  String email;
+  String fullName;
+  String dateJoined;
+  // Really sometimes absent in /register, but we normalize that away.
+  // Background: https://chat.zulip.org/#narrow/stream/412-api-documentation/topic/.60is_active.60.20in.20.60.2Fregister.60.20response/near/1371603
+  bool isActive;
+  bool isOwner;
+  bool isAdmin;
+  bool isGuest;
+  bool? isBillingAdmin; // TODO(server-5)
+  bool isBot;
+  int? botType; // TODO enum
+  int? botOwnerId;
+  int role; // TODO enum
+  String timezone;
+  String? avatarUrl; // TODO distinguish null from missing https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/flutter.3A.20omitted.20vs.2E.20null.20in.20JSON/near/1551759
+  int avatarVersion;
+  final Map<int, ProfileFieldUserData>? profileData;
```

Comment:
Would it be just as good to do this instead:

```dart
late final Map<int, ProfileFieldUserData> profileData = {};
```

?

From the doc:

> When you mark a variable as `late` but initialize it at its declaration, then the initializer runs the first time the variable is used. This lazy initialization is handy in a couple of cases:
> 
> - The variable might not be needed, and initializing it is costly.
> - […]

Reply:
That version would still mean that we'd end up allocating these if we ever look at the users' profile data. I'm not sure we necessarily would end up ever looking en masse at all or most users' profile data — but if we did, that'd be on the face of it a pretty innocent bit of code, and so this version of the field would be laying a performance trap that we could later walk into.

For that reason, better to have an explicit null. Then the consuming code will just have to write `?.` for method calls on it and so on (and code that's updating it will need to write `??= {}`); the type-checker means we can't forget to do those, and the performance characteristics will always be straightforward.